### ISSUE TITLE
Decommission stopped and pool is back to active

### ISSUE THREAD

**Issue by olevitt** [minio/minio#19249]
Hi,  

Sorry to bother again but there is a followup to #19179 .
After deleting leftover `rebalance.bin` and starting a decommission, everything went well for a few hours (`mc admin decommission status` reported pool as draining and `mc admin decommission status --json` reported progress) and now, few hours after that, it seems there is no trace of the decommission process and the pool is back to active (which, according to https://min.io/docs/minio/linux/operations/install-deploy-manage/decommission-server-pool.html is never supposed to happen).  
So I'm kind of lost here on what could have happen.  
The cluster seems to be fine, data from all 3 pools is accessible, no outage reported but I'm puzzled on the decommission status and what it implies for the future (should I try the decommission again ?) and think there is a possible bug (especially considering #19179) so I'm opening this issue.

Yesterday :  
```
Decommission started successfully for https://minio-datanode-pool-0-{0...3}.minio-datanode-hl.minio-datanode.svc.cluster.local/export{0...17}.
```
Pool was marked as draining : 
```
gon@laboitemagique:~$ mc admin decommission status sspcloud
┌─────┬─────────────────────────────────────────────────────────────────────────────────────────────────────────┬────────────────────────┬──────────┐
│ ID  │ Pools                                                                                                   │ Raw Drives Usage       │ Status   │
│ 1st │ https://minio-datanode-pool-0-{0...3}.minio-datanode-hl.minio-datanode.svc.cluster.local/export{0...17} │ 20.7% (total: 52 TiB)  │ Draining │
│ 2nd │ https://minio-datanode-pool-1-{0...3}.minio-datanode-hl.minio-datanode.svc.cluster.local/export{0...17} │ 21.1% (total: 105 TiB) │ Active   │
│ 3rd │ https://minio-datanode-pool-2-{0...3}.minio-datanode-hl.minio-datanode.svc.cluster.local/export{0...11} │ 5.9% (total: 79 TiB)   │ Active   │
```

Today :  
```
gon@laboitemagique:~$ mc admin decommission status sspcloud
┌─────┬─────────────────────────────────────────────────────────────────────────────────────────────────────────┬────────────────────────┬────────┐
│ ID  │ Pools                                                                                                   │ Raw Drives Usage       │ Status │
│ 1st │ https://minio-datanode-pool-0-{0...3}.minio-datanode-hl.minio-datanode.svc.cluster.local/export{0...17} │ 14.9% (total: 52 TiB)  │ Active │
│ 2nd │ https://minio-datanode-pool-1-{0...3}.minio-datanode-hl.minio-datanode.svc.cluster.local/export{0...17} │ 22.6% (total: 105 TiB) │ Active │
│ 3rd │ https://minio-datanode-pool-2-{0...3}.minio-datanode-hl.minio-datanode.svc.cluster.local/export{0...11} │ 7.8% (total: 79 TiB)   │ Active │
└─────┴─────────────────────────────────────────────────────────────────────────────────────────────────────────┴────────────────────────┴────────┘
```
and 
```
gon@laboitemagique:~$ mc admin decommission status sspcloud --json
[
    {
        "id": 0,
        "cmdline": "https://minio-datanode-pool-0-{0...3}.minio-datanode-hl.minio-datanode.svc.cluster.local/export{0...17}",
        "lastUpdate": "2024-03-11T20:24:22.129770094Z",
        "decommissionInfo": {
            "startTime": "0001-01-01T00:00:00Z",
            "startSize": 0,
            "totalSize": 57583566716928,
            "currentSize": 49027129098240,
            "complete": false,
            "failed": false,
            "canceled": false,
            "objectsDecommissioned": 0,
            "objectsDecommissionedFailed": 0,
            "bytesDecommissioned": 0,
            "bytesDecommissionedFailed": 0
        }
    },
    {
        "id": 1,
        "cmdline": "https://minio-datanode-pool-1-{0...3}.minio-datanode-hl.minio-datanode.svc.cluster.local/export{0...17}",
        "lastUpdate": "2023-05-07T13:57:33.781479532Z",
        "decommissionInfo": {
            "startTime": "0001-01-01T00:00:00Z",
            "startSize": 0,
            "totalSize": 115134777262080,
            "currentSize": 89061980131328,
            "complete": false,
            "failed": false,
            "canceled": false,
            "objectsDecommissioned": 0,
            "objectsDecommissionedFailed": 0,
            "bytesDecommissioned": 0,
            "bytesDecommissionedFailed": 0
        }
    },
    {
        "id": 2,
        "cmdline": "https://minio-datanode-pool-2-{0...3}.minio-datanode-hl.minio-datanode.svc.cluster.local/export{0...11}",
        "lastUpdate": "2024-02-29T10:08:25.769920098Z",
        "decommissionInfo": {
            "startTime": "0001-01-01T00:00:00Z",
            "startSize": 0,
            "totalSize": 86374960054272,
            "currentSize": 79636222541824,
            "complete": false,
            "failed": false,
            "canceled": false,
            "objectsDecommissioned": 0,
            "objectsDecommissionedFailed": 0,
            "bytesDecommissioned": 0,
            "bytesDecommissionedFailed": 0
        }
    }
]
```

## Your Environment

* Minio version : `2024-03-10T02:53:48Z`
* Tenant configuration see #19179

**Comment by harshavardhana** [comment]
Looks like you have a very flaky setup @olevitt

**Comment by harshavardhana** [comment]
do you have server logs during this time period from the first node of the first pool?

**Comment by harshavardhana** [comment]
In your logs grep for "saving pool.bin for"

**Comment by olevitt** [comment]
Hi ! 

> do you have server logs during this time period from the first node of the first pool?

I only have logs for the latest start of the pods. Sadly it doesnt include the decommission process as we restarted the pods since, sorry :( 
No trace of `saving pool.bin for` in this run on any of the pods.

**Comment by harshavardhana** [comment]
did you have any pod restarts during the period where you saw this turn the pool active again? @olevitt

**Comment by olevitt** [comment]
Here is what happened (sorry I don't have logs for the previous runs) :  
- We got hit by #19179, cleaned up `rebalance.bin` and restarted the pods
- March 11th 13:31(Paris Time) : We started decommission
- March 11th 16:47 : decommission was still in progress with about 10% of bytes drained, `"objectsDecommissionedFailed": 1,`
- March 12th 09:16 : decommission had stopped (probably had stopped before that point but that's thé first time I checked the status since the previous day), pool was back active. There was no pod restart during the decommission period.
- After that : we tried restarting pods with no change

Should I try restarting the decommission process and sée how it goes ? Or is that a bad Idea ?

**Comment by harshavardhana** [comment]
It's actually very simple thing and already fixed it's a UI issue with mc not related to pool becoming active again.

This is fixed and should report correctly from the next release.

Fixed via #19300
